nucleus/common/internal-api/pom.xml

```diff
@@ -129,6 +130,18 @@
             <artifactId>logging-annotation-processor</artifactId>
             <optional>true</optional>
         </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>1.16.14</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.weld</groupId>
```

Comment:
I don't see any reason why I can't use weld SPI here...
Do you have a specific objection?